psycopg/psycopg/connection_async.py

```diff
@@ -34,6 +34,17 @@
 logger = logging.getLogger("psycopg")
 
 
+class AsyncPipeline(BasePipeline):
+    """Handler for async connection in pipeline mode."""
+
+    async def status(self) -> PipelineStatus:
```

Comment:
Why is this an async method? `PQpipelineStatus` is only a C member lookup, it doesn't do I/O. I think we should make `status` a property of `BasePipeline`.

Actually, looking at the `pq` wrappers, I think that `pq.pipeline_status` might be a property and we could actually wrap all the three status/enter/exit under a writable property. I've looked at the source code and none of them should do any I/O.

Reply:
You're totally right. I'm still kind of looking for the way in libpq API so my reasoning is sometimes wrong. Thanks for the suggestion, I'll apply that pattern.